micrometer-observation-test/src/main/java/io/micrometer/observation/tck/TestObservationRegistry.java

```diff
@@ -179,6 +191,37 @@ Observation.Context getContext() {
             return this.context;
         }
 
+        /**
+         * Stores an {@link Observation.Event} in this context.
+         * @param event the event to store
+         */
+        public void addEvent(Observation.Event event) {
+            this.contextEvents.add(event);
+        }
+
+        /**
+         * Check if an {@link Observation.Event} with the given name was stored in this
+         * context.
+         * @param name name of the event to check
+         * @return {@code true} if an event was stored under the given name
+         */
+        public boolean hasEvent(String name) {
+            return this.contextEvents.stream().anyMatch(event -> event.getName().equals(name));
+        }
+
+        /**
+         * Check if an {@link Observation.Event} with the given name and contextual name
+         * was stored in this context.
+         * @param name name of the event to check
+         * @param contextualName contextual name of the event to check
+         * @return {@code true} if an event was stored under the given name and contextual
+         * name
+         */
+        public boolean hasEvent(String name, String contextualName) {
```

Comment:
I think the methods added here to the TestContext do not need to be public. Could we get away with them being package private? We don't want to add unnecessary public API in case we want to change it, and users should be using the API on TestObservationRegistryAssert, right?

Reply:
Good catch - I'll get that fixed.